src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/IPropertyExtensions.cs

```diff
@@ -28,12 +28,12 @@ internal static IPropertyMetadata ToPropertyMetadata(this IProperty property, Ty
             propertyMetadata.IsForeignKey = property.IsForeignKey();
 
             propertyMetadata.IsEnum = property.ClrType.GetTypeInfo().IsEnum;
-            propertyMetadata.IsReadOnly = property.IsReadOnlyBeforeSave;
+            propertyMetadata.IsReadOnly = property.BeforeSaveBehavior != PropertyValueBehavior.UseValue;
 
             //TODO: RequiresValueGenerator() is an internal EF Core method.
             // Need to confirm with EF if this is ok to use. 
             propertyMetadata.IsAutoGenerated = ((property.RequiresValueGenerator()
-                    || property.IsStoreGeneratedAlways)
+                    || property.BeforeSaveBehavior != PropertyValueBehavior.UseValue)
                 && (property.ValueGenerated != ValueGenerated.Never));
```

Comment:
Fixed in https://github.com/aspnet/Scaffolding/commit/c2847addc9561213f3e59a43c77fc48bcad9da82